### PR TITLE
[installer, gitpod-db] Introduce database.ssl.ca

### LINKED ISSUE #12012: Support SSL configurations of mysql connections
## Is your feature request related to a problem? Please describe
Currently gitpod-db service only [takes basic credentials](https://github.com/gitpod-io/gitpod/blob/2fd3dd3586ff4e6ed23669e7be5827bbc27eeacf/components/gitpod-db/src/config.ts#L29-L38) which is counter to many corporate requirements around best practices for access. Without it users have to raise an exception to be allowed to install it.

The [mysql](https://github.com/mysqljs/mysql#ssl-options) library gitpod-db uses exposes a configurable ssl setting, including RDS specific configuration that loads the AmazonAWS certificates.

The only other change appears to be in our db migration job, which would need to be modifiable in cases where SSL is used, since it is [performing a mysql cli command](https://github.com/gitpod-io/gitpod/blob/main/install/installer/pkg/components/database/init/job.go#L54-L72).

## Describe the behaviour you'd like
Allow the user to enable at minimum `--ssl-mode=VERIFY_IDENTITY` and `--ssl-ca=custom-ca.crt` level of options to their mysql configuration part of the Gitpod installation.

If it is an RDS mysql instance, the user should be able to specify that.

### PR DESCRIPTION
**Pull request by geropl**
## Description
This allows Gitpod to directly created SSL encrypted connections to MySQL databases.

For testing :point_down: there are two environments:
 - preview on this branch (w/o SSL)
 - the AWS release test instance (deployed from this branch with the `deps: external` flag) with external RDS with SSL.
   - It uses the default jobs config, with two changes:
     - a `secret` named `database-ssl` and an attribute `ca.crt` that contains the cert (chain) to validate the RDS certificate
     - the installer config slightly adjust to:
     ```
      database:
        inCluster: false
        external:
            ...
        ssl:
          customCa:
            kind: secret
            name: database-ssl
      ```

## Related Issue(s)

Fixes #12012

## How to test
- sign in and start a workspace in the [preview env](https://gpl-12012-ssl.preview.gitpod-dev.com/workspaces) (w/o SSL): works as expected
- sign in and ~start a workspace~* got to [notification settings](https://release-aws.tests.gitpod-self-hosted.com/notifications) in the [AWS release test install](https://release-aws.tests.gitpod-self-hosted.com/workspaces) and change them (WITH SSL required): works as well! :heavy_check_mark: 

*: there is an odd content-init error I get on the AWS realease test env. @nandajavarma any idea where that might come from? :thinking: 

## Release Notes

```release-note
Allow specifying CA certificate to configure SSL secured database connections
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by geropl** [comment]
@mrzarquon @mrsimonemms Could you help me test this change? I think the quickest would be if you could point me to an AWS installation and show me how to configure it with the certs from [here](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.SSL.html) &deploy it...? :thinking:

**Comment by mrsimonemms** [comment]
@geropl looks good, but it could do with a [cluster validation check](https://github.com/gitpod-io/gitpod/blob/gpl/12012-ssl/install/installer/pkg/config/v1/validation.go#L122) to ensure the secret is uploaded and valid.

It'll be something like:

```go
if cfg.Config.Database.SSL != nil && cfg.Config.Database.SSL.CustomCA != nil {
	secretName := cfg.Config.Database.SSL.CustomCA.Name
	res = append(res, cluster.CheckSecret(secretName, cluster.CheckSecretRequiredData("<some key1>", "<some key2>")))
}
```

The `<some key1>` `<some key2>` are the keys you want to ensure are present in the secret.

File: install/installer/pkg/components/database/init/job.go
Comment (mrsimonemms) [inline install/installer/pkg/components/database/init/job.go:91]: Does this need a space after the `%s`, eg.
```suggestion
							fmt.Sprintf("mysql -h $DB_HOST --port $DB_PORT -u $DB_USERNAME -p$DB_PASSWORD %s < /db-init-scripts/init.sql", sslOptions),
```

**Review by mrsimonemms — APPROVED** [review]
One question, but happy with the change

File: install/installer/pkg/components/database/init/job.go
Comment (geropl) [reply install/installer/pkg/components/database/init/job.go:91]: It does not need to be there, so I'd opt for keeping the commits clean(er). :+1: 

(I wish there was a update-and-integrate-into-last-commit button here :roll_eyes: )

Comment (mrsimonemms) [reply install/installer/pkg/components/database/init/job.go:91]: > It does not need to be there, so I'd opt for keeping the commits clean(er). +1

Fine by me

> (I wish there was a update-and-integrate-into-last-commit button here roll_eyes )

Yes, that would be nice. Oh well.

File: components/gitpod-db/go/conn.go
Comment (easyCZ) [inline components/gitpod-db/go/conn.go:26]:
```suggestion
	CACert string
```

Comment (easyCZ) [reply components/gitpod-db/go/conn.go:26]: Just feels we don't need to push the "custom" part into this

File: components/public-api-server/pkg/server/server.go
Comment (easyCZ) [inline components/public-api-server/pkg/server/server.go:52]: Likewise here, I'd drop the custom.

**Review by easyCZ — APPROVED** [review]
LGTM % modulo dropping `custom` from the references. All options are custom, there shouldn't be a need to duplicate that in the name.

/hold

File: components/public-api-server/pkg/server/server.go
Comment (geropl) [reply components/public-api-server/pkg/server/server.go:52]: Yeah, good point. So far I have been torn we because I'd have to partially retest. But I think you're right. :100:

Comment (geropl) [reply components/public-api-server/pkg/server/server.go]: Ok. Done, and re-tested: works.

**Comment by geropl** [comment]
/unhold

[ref](https://github.com/gitpod-io/gitpod/pull/15320#discussion_r1049781945)

**Pull request merged**
